app/scripts/metamask-controller.js

```diff
@@ -440,22 +440,21 @@ module.exports = class MetamaskController extends EventEmitter {
   // Identity Management
   //
 
-  newUnapprovedTransaction (txParams, cb) {
+  async newUnapprovedTransaction (txParams) {
     log.debug(`MetaMaskController newUnapprovedTransaction ${JSON.stringify(txParams)}`)
-    const self = this
-    self.txController.addUnapprovedTransaction(txParams, (err, txMeta) => {
-      if (err) return cb(err)
-      self.sendUpdate()
-      self.opts.showUnapprovedTx(txMeta)
-      // listen for tx completion (success, fail)
-      self.txController.once(`${txMeta.id}:finished`, (completedTx) => {
+    const txMeta = await this.txController.addUnapprovedTransaction(txParams)
+    this.sendUpdate()
+    this.opts.showUnapprovedTx(txMeta)
```

Comment:
instead of calling it here, we should listen for the `metamaskController.txController.on('newTx')` or w/e the event is